### ISSUE TITLE
Lint instance fields that could be local variables

### ISSUE THREAD

**Issue by alexeieleusis** [dart-lang/linter#801]
I don't fully know how to identify them but a good candidate is a parameter of a constructor or method that is only used there.

**Comment by bwilkerson** [comment]
You'd want to look for any field that (a) is only referenced in a single method, and (b) whose state is not dependent on whether or not the method has previously been invoked. The first is easy to figure out, but the second is a bit harder. Basically, it means that either the field is assigned a value before the value is read (along all possible flow paths) or that the field is always assigned the same value before the method exits (and before the first invocation of the method). Both conditions require some form of flow analysis, which hasn't yet been implemented.

**Comment by srawlins** [comment]
Wouldn't this only apply to private fields?

**Comment by bwilkerson** [comment]
Yes. That's another complication. Public fields can be accessed and possibly set from anywhere, and lints only operate on a single compilation unit at a time.
